From 17377879627da2ae809bfc7d2ddd2ff75a1707c0 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Tue, 29 Apr 2014 17:07:39 -0400
Subject: [Documentation] mei-amt-version: remove unneeded call of mei_deinit()

Message-id: <1398791260-24689-3-git-send-email-prarit@redhat.com>
Patchwork-id: 79231
O-Subject: [RHEL7.1 PATCH BZ 1090078 2/3] Documentation/misc-devices/mei/mei-amt-version.c: remove unneeded call of mei_deinit()
Bugzilla: 1090078
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Rik van Riel <riel@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1090078

commit aa98a28563d2abcd904d07fd5e54553290c85f91
Author: Paul Bolle <pebolle@tiscali.nl>
Date:   Fri Jan 3 14:44:19 2014 -0800

    Documentation/misc-devices/mei/mei-amt-version.c: remove unneeded call of mei_deinit()

    Building mei-amt-version.o triggers a GCC warning:
        Documentation/misc-devices/mei/mei-amt-version.c: In function 'main':
        Documentation/misc-devices/mei/mei-amt-version.c:103:5: warning: 'acmd.fd' is used uninitialized in this function [-Wuninitialized]
          if (cl->fd != -1)
             ^
        Documentation/misc-devices/mei/mei-amt-version.c:443:21: note: 'acmd.fd' was declared here
          struct amt_host_if acmd;
                             ^

    GCC is correct. See, the call chain that GCC detects must be
        main()
            amt_host_if_init()
                mei_init()
                    mei_deinit()

    But when we enter mei_deinit() struct amt_host_if acmd is still
    unitialized. That makes the test for (effectively) amt_host_if->mei_cl->fd
    bogus.

    But it turns out that call of mei_deinit() isn't needed at all. All of
    the members of mei_cl will be set later in mei_init() and none will be
    used before they are set. So we can simply drop this call of
    mei_deinit().

    Signed-off-by: Paul Bolle <pebolle@tiscali.nl>
    Cc: Tomas Winkler <tomas.winkler@intel.com>
    Reported-by: David Howells <dhowells@redhat.com>
    Acked-by: David Howells <dhowells@redhat.com>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Cc: David Howells <dhowells@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/Documentation/misc-devices/mei/mei-amt-version.c b/Documentation/misc-devices/mei/mei-amt-version.c
index 49e4f77..57d0d87 100644
--- a/Documentation/misc-devices/mei/mei-amt-version.c
+++ b/Documentation/misc-devices/mei/mei-amt-version.c
@@ -115,8 +115,6 @@ static bool mei_init(struct mei *me, const uuid_le *guid,
  struct mei_client *cl;
  struct mei_connect_client_data data;
 
- mei_deinit(me);
-
  me->verbose = verbose;
 
  me->fd = open("/dev/mei", O_RDWR);
-- 
1.7.1